src/jloptions.c

```diff
@@ -0,0 +1,431 @@
+
+    // code generation options
+    " --compile={yes|no|all|min}Enable or disable JIT compiler, or request exhaustive compilation\n"
+    " -C, --cpu-target <target> Limit usage of cpu features up to <target>\n"
+    " -O, --optimize={0,1,2,3}  Set the optimization level (default 2 if unspecified or 3 if specified as -O)\n"
+    " --inline={yes|no}         Control whether inlining is permitted (overrides functions declared as @inline)\n"
+    " --check-bounds={yes|no}   Emit bounds checks always or never (ignoring declarations)\n"
+    " --math-mode={ieee,fast}   Disallow or enable unsafe floating point optimizations (overrides @fastmath declaration)\n\n"
+
+    // error and warning options
+    " --depwarn={yes|no|error}  Enable or disable syntax and method deprecation warnings (\"error\" turns warnings into errors)\n\n"
+
+    // compiler output options
+    " --output-o name           Generate an object file (including system image data)\n"
+    " --output-ji name          Generate a system image data file (.ji)\n"
+    " --output-bc name          Generate LLVM bitcode (.bc)\n"
+    " --output-incremental=no   Generate an incremental output file (rather than complete)\n\n"
+
+    // instrumentation options
+    " --code-coverage={none|user|all}, --code-coverage\n"
+    "                           Count executions of source lines (omitting setting is equivalent to \"user\")\n"
+    " --track-allocation={none|user|all}, --track-allocation\n"
+    "                           Count bytes allocated by each source line\n\n"
+
+    "Deprecated options:\n"
+    " -F                        Load ~/.juliarc (deprecated, use --startup-file=yes)\n"
+    " -f, --no-startup          Don't load ~/.juliarc (deprecated, use --startup-file=no)\n"
+    " -P, --post-boot <expr>    Evaluate <expr>, but don't disable interactive mode (deprecated, use -i -e instead)\n"
+    " --no-history-file         Don't load history file (deprecated, use --history-file=no)\n";
+
+JL_DLLEXPORT void jl_parse_opts(int *argcp, char ***argvp)
+{
+    enum { opt_machinefile = 300,
+           opt_color,
+           opt_history_file,
+           opt_no_history_file,
+           opt_startup_file,
+           opt_compile,
+           opt_code_coverage,
+           opt_track_allocation,
+           opt_check_bounds,
+           opt_output_bc,
+           opt_depwarn,
+           opt_inline,
+           opt_math_mode,
+           opt_worker,
+           opt_bind_to,
+           opt_handle_signals,
+           opt_output_o,
+           opt_output_ji,
+           opt_use_precompiled,
+           opt_use_compilecache,
+           opt_incremental
+    };
+    static char* shortopts = "+vhqFfH:e:E:P:L:J:C:ip:O:";
+    static struct option longopts[] = {
+        // exposed command line options
+        // NOTE: This set of required arguments need to be kept in sync
+        // with the required arguments defined in base/client.jl `process_options()`
```

Comment:
there's a corresponding comment in `base/client.jl` that needs to be changed to reflect the move